code/data_structures/src/list/circular_linked_list/circular_linkedlist.c

```diff
@@ -0,0 +1,289 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct node{
+    int data;
+    struct node *link;
+};
+
+struct node *head=NULL;
```

Comment:
here I am also finding reverse so head is defined as global

Reply:
Still, pass all of your variables as parameters. Most C linked list implementations pass `head` for all operations. Global variables are considered evil for a reason.